spec/unit/blockscore/question_set_spec.rb

```diff
@@ -1,42 +1,128 @@
-        end
+        it { is_expected.to be_persisted }
+        it { expect(question_set.class).to be BlockScore::QuestionSet }
+      end
+
+      context 'invaild person' do
+        let (:person_id) { create(:invalid_person).id }
+
+        it { expect { BlockScore::QuestionSet.create(person_id: person_id) }.to raise_error BlockScore::InvalidRequestError }
       end
+    end
+
+    pending '.find' do
+      context 'valid question set id' do
+        let(:question_set_id)  { create(:question_set).id }
+        subject(:question_set) { BlockScore::QuestionSet.find(question_set_id) }
+
+        it { is_expected.to be_persisted }
+        it { expect(question_set.questions).not_to be_empty }
+      end
+
+      context 'invalid person id' do
+        let(:question_set_id) { 'f3a243ddc8075a6acd603b7758d05b3c' }
+
+        it { expect { BlockScore::QuestionSet.find(question_set_id) }.to raise_error BlockScore::RecordNotFound }
+      end
+    end
+
+    describe '.retrieve' do
+      let(:question_set_id)  { create(:question_set).id }
+      subject(:question_set) { BlockScore::QuestionSet.retrieve(question_set_id) }
+
+      it { is_expected.to be_persisted }
+      it { expect(question_set.questions).not_to be_empty }
+      it { expect(question_set.class).to be BlockScore::QuestionSet }
+    end
+
+    describe '#score' do
+      context 'correct answers' do
+        subject(:question_set) { create(:question_set) }
+        before do
+          answers = []
 
-      context 'when scoring' do
-        let(:answers) do
-          [
-            { question_id: 1, answer_id: 1 },
-            { question_id: 2, answer_id: 1 },
-            { question_id: 3, answer_id: 1 },
-            { question_id: 4, answer_id: 1 },
-            { question_id: 5, answer_id: 1 }
-          ]
+          question_set.questions.each do |question|
+            question.answers.each do |answer|
+              if ['309 Colver Rd', '812', 'Jasper', '49230', 'None Of The Above'].include? answer.answer
+                answers << { question_id: question.id, answer_id: answer.id }
+                break
+              end
+            end
+          end
+
+          question_set.score(answers)
         end
 
-        it 'score call does request' do
-          qs = person.question_sets.create
-          qs.score(answers)
-          assert_requested(api_stub, times: 3)
+        it { expect(question_set.score).to eq 100.0 }
+      end
+
+      context 'incorrect answers' do
+        subject(:question_set) { create(:question_set) }
+        before do
+          answers = []
+
+          question_set.questions.each do |question|
+            question.answers.each do |answer|
+              if ['309 Colver Rd', '812', 'Jasper', '49230', 'None Of The Above'].include? answer.answer
+                answers << { question_id: question.id, answer_id: (answer.id + 1) % 5 }
+                break
+              end
+            end
+          end
```

Comment:
Might want to reduce the duplication between the different contexts here.  Maybe just pull the common part into a module